src/txrebroadcast.h

```diff
@@ -41,13 +41,19 @@ class TxRebroadcastHandler
     /** Remove transaction entry from the attempt tracker.*/
     void RemoveFromAttemptTracker(const CTransactionRef& tx);
 
+    /** Test only */
+    void UpdateAttempt(const uint256& wtxid, const int count, const std::chrono::microseconds last_attempt_time);
+
+    /** Test only */
+    bool CheckRecordedAttempt(const uint256& wtxid, const int expected_count, const std::chrono::microseconds expected_timestamp) const;
```

Comment:
In 87d6466380737c2d05932d4af3879950982e61c4 [test] Add unit test for rebroadcast attempt logic 

I'm not sure what difference it would make, but why don't these require `m_rebroadcast_mutex` instead of doing the locking internally?

Reply:
just so I can keep `m_rebroadcast_mutex` as a private member